v1/contracts/Voting.sol

```diff
@@ -33,9 +99,109 @@ contract Voting {
      * committer can reveal their vote.
      */
     function revealVote(bytes32 identifier, uint time, int price, int salt) external {
-        VoteInstance storage voteInstance = requests[identifier][time];
+        VoteInstance storage voteInstance = priceResolutions[identifier][time].votes[currentRoundNumber];
         require(keccak256(abi.encode(price, salt)) == voteInstance.committedHashes[msg.sender],
                 "Committed hash doesn't match revealed price and salt");
         delete voteInstance.committedHashes[msg.sender];
     }
+
+    /**
+     * @notice Enqueues a request (if a request isn't already present) for the given `identifier`, `time` pair.
+     * @dev Returns the time at which the user should expect the price to be resolved.
+     */
+    function requestPrice(bytes32 identifier, uint time) external returns (uint expectedTime) {
+        uint priceResolutionRound = priceResolutions[identifier][time].lastVotingRound;
```

Comment:
Is the plan to add verification at the beginning of `requestPrice` and `getPrice` to check if the caller paid enough in fees? If so, does this contract have a reference to the Store?

Reply:
We discussed price request spam prevention. Do we track the contracts with an active price request in order to prevent additional price requests? Or instead, do we protect against spam using some whitelist mechanism?

Reply:
The idea (at least for now) is that we shouldn't have to worry about either of these because of whitelisting.

The biggest issue with tracking and checking these on the Oracle side rather than the contract side is that I think it would add *more* complexity to the whitelisted contracts because they would need to build in a way to respond to rejected price requests from the oracle.

If we didn't want to build rejection handling into the financial contracts, we would need to make sure that they never violated the Oracle rules, which is the same thing we're doing now. So that wouldn't really reduce complexity in the contract, it would only increase complexity in the Oracle because we'd be double checking all these conditions.

Does that make sense?

Reply:
Gotcha, yeah, we did have some of these conversations earlier. Whitelisting without any checks on either the oracle or the financial contract sounds good for this iteration, since we're sticking to the existing requirements.